docs/DesignDocs/SealingAPI.md

```diff
@@ -0,0 +1,1171 @@
+the *Accommodative* requirement. Option 2 is our choice.
+
+Now that `key_info` has been passed to `oe_seal()` and it has to be persisted
+for `oe_unseal()`, it's a natural idea to embed `key_info` into the sealed
+blob. Combined with the considerations above we come to the declarations of
+`oe_seal()` and `oe_unseal()` that are shown below. Meanings of the function
+parameters should be intuitive. More details can be found in
+[oe_seal()](#oe_seal()) and [oe_unseal()](#oe_unseal()) later in this document.
+
+```C
+oe_result_t oe_seal(
+    const uint8_t* key_info,
+    size_t key_info_size,
+    const uint8_t* plaintext,
+    size_t plaintext_size,
+    const uint8_t* additional_data,
+    size_t additional_data_size,
+    uint8_t **blob,
+    size_t *blob_size);
+
+oe_result_t oe_unseal(
+    uint8_t* blob,
+    size_t blob_size,
+    uint8_t** plaintext,
+    size_t* plaintext_size,
+    uint8_t** additional_data,
+    size_t* additional_data_size);
+```
+
+The code snippet below shows how to encrypt data (i.e., `my_data_size` bytes
+starting at `my_data`) using `oe_seal()`.
+
+```C
+uint8_t* blob;
+size_t blob_size;
+result = oe_seal(key_info,
+                 key_info_size,
+                 my_data,       // plaintext
+                 my_data_size,  // plaintext_size
+                 NULL,          // additional_data
+                 0,             // additional_data_size
+                 &blob,
+                 &blob_size);
+id (result != OE_OK)
+    // Handle encryption error...
+else
+{
+    // Persist blob...
+
+    oe_free(blob); // Free blob when done
+}
+```
+
+Please note that `my_data` above will be both encrypted and
+integrity-protected. But if only integrity needs protection, one could instead
+pass `my_data` and `my_data_size` as `additional_data` and
+`additional_data_size`, respectively.
+
+```C
+uint8_t* blob;
+size_t blob_size;
+result = oe_seal(key_info,
+                 key_info_size,
+                 NULL,          // plaintext
+                 0,             // plaintext_size
+                 my_data,       // additional_data
+                 my_data_size,  // additional_data_size
+                 &blob,
+                 &blob_size);
+id (result != OE_OK)
+    // Handle encryption error...
+else
+{
+    // Persist blob...
+
+    oe_free(blob); // Free blob when done
+}
+```
+
+Please note that the blob format is proprietary so there's no architectural way
+to extract `additional_data` or `additional_data_size` without unsealing the
+blob. That said, the only way to recover `my_data` (and `my_data_size`) is
+still via `oe_unseal()`, regardless of whether it was encrypted or not. In other words,
+the two code snippets above are identical from a functional perspective but
+differ in security. Therefore, developers are strongly encouraged to use
+encryption for comprehensive protection.
+
+`oe_unseal()` decrypts a sealed blob **in-place**. The code snippet below shows
```

Comment:
```suggestion
The code snippet below shows
```
It should not require decryption in place, which would limit applicability.  This also implies you should add free in the sample, just like oe_seal.